paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/sink/SupportsRowLevelOperationFlinkTableSink.java

```diff
@@ -185,11 +185,18 @@ private void validateDeletable() {
                             table.getClass().getName()));
         }
 
-        MergeEngine mergeEngine = CoreOptions.fromMap(table.options()).mergeEngine();
-        if (mergeEngine != DEDUPLICATE) {
-            throw new UnsupportedOperationException(
-                    String.format("Merge engine %s can not support batch delete.", mergeEngine));
+        CoreOptions coreOptions = CoreOptions.fromMap(table.options());
+        if (coreOptions.mergeEngine() == DEDUPLICATE
+                || coreOptions.ignoreDelete()
```

Comment:
If its `ignoreDelete`, do we have to allow it execute `delete from`?

Reply:
Yes, If its ignoreDelete, it should not be allowed to execute `delete from`.